src/httprpc.cpp

```diff
@@ -256,8 +278,11 @@ static bool InitRPCAuthentication()
         for (const std::string& rpcauth : gArgs.GetArgs("-rpcauth")) {
             std::vector<std::string> fields;
             boost::split(fields, rpcauth, boost::is_any_of(":$"));
-            if (fields.size() == 3) {
+            if (fields.size() >= MIN_RPCAUTH_VALUES) {
                 g_rpcauth.push_back(fields);
+                if (fields.size() > MAX_RPCAUTH_VALUES) {
+                    LogPrintf("Unrecognised -rpcauth parameters for username '%s'. User will not be able to authenticate.\n", SanitizeString(fields[0]));
+                }
             } else {
                 LogPrintf("Invalid -rpcauth argument.\n");
                 return false;
```

Comment:
I think you are overcomplicating this. No need for a +53-7 patch when this line could be removed in a -1+0 patch to achieve the same user experience

Reply:
#20550